enterprise/server/ociregistry/ociregistry.go

```diff
@@ -173,6 +173,83 @@ func (r *registry) handleV2Request(ctx context.Context, w http.ResponseWriter, i
 	}
 }
 
+func makeUpstreamRequest(ctx context.Context, method, acceptHeader, authorizationHeader string, ociResourceType ocipb.OCIResourceType, ref gcrname.Reference) (*http.Response, error) {
+	var path string
+	switch ociResourceType {
+	case ocipb.OCIResourceType_BLOB:
+		path = "/v2/" + ref.Context().RepositoryStr() + "/blobs/" + ref.Identifier()
+	case ocipb.OCIResourceType_MANIFEST:
+		path = "/v2/" + ref.Context().RepositoryStr() + "/manifests/" + ref.Identifier()
+	case ocipb.OCIResourceType_UNKNOWN:
+		return nil, fmt.Errorf("unknown OCI resource type, expected blobs or manifests")
+	}
+	u := url.URL{
+		Scheme: ref.Context().Scheme(),
+		Host:   ref.Context().RegistryStr(),
+		Path:   path,
+	}
+	upreq, err := http.NewRequest(method, u.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not make %s request to upstream registry '%s': %s", method, u.String(), err)
+	}
+	if acceptHeader != "" {
+		upreq.Header.Set(headerAccept, acceptHeader)
+	}
+	if authorizationHeader != "" {
+		upreq.Header.Set(headerAuthorization, authorizationHeader)
+	}
+	upresp, err := http.DefaultClient.Do(upreq.WithContext(ctx))
+	return upresp, err
+}
+
+func parseContentLengthHeader(contentLengthHeader string) (bool, int64, error) {
+	hasContentLength := contentLengthHeader != ""
+	if hasContentLength {
+		contentLength, err := strconv.ParseInt(contentLengthHeader, 10, 64)
+		if err != nil {
+			return false, 0, fmt.Errorf("could not parse %s header (value '%s'): %s", headerContentLength, contentLengthHeader, err)
+		}
+		if contentLength < 0 {
+			return false, 0, fmt.Errorf("%s header must be 0 or greater, received value '%s'", headerContentLength, contentLengthHeader)
+		}
+		return true, contentLength, nil
+	}
+	return false, 0, nil
+}
+
+func parseDockerContentDigestHeader(dockerContentDigestHeader string) (bool, *gcr.Hash, error) {
```

Comment:
You can keep the variable names short where their meaning is obvious. The function is called `parseDockerContentDigestHeader` so it's ok to call the variable something simple like 'header' or 'value'